testsuite/MDAnalysisTests/core/test_segmentgroup.py

```diff
@@ -109,3 +111,16 @@ def test_set_segids_many():
 def test_atom_order(universe):
     assert_equal(universe.segments.atoms.indices,
                  sorted(universe.segments.atoms.indices))
+
+class TestSegmentGroupPickle(object):
+    """Test SegmentGroup pickling support."""
+    @pytest.fixture()
+    def universe(self):
+        return mda.Universe(TPR, XTC)
```

Comment:
minor quibble, but if you use `Universe.empty()` it'll make the tests a little faster.  it's not a big deal, but over 1,000 tests it adds up slowly

Reply:
Sorry, don't get it. Say I do:
```
seg_group = mda.Universe.empty(10).segments
seg = pickle.loads(pickle.dumps(seg_group))
```

how do I compare them? And would that be a useful test?

Reply:
you could check that the new segments have the same universe as the old one:
```python
u = mda.Universe.empty()

segs = u.segments

new = pickle.loads(...)

assert new.universe is u
```

I think maybe you're right, a better test would have dummy data inside the Universe...
```python
u = mda.Universe.empty()
u.add_TopologyAttr('segids', values=['A', 'B', etc]

... some pickling

assert new.segids == ['A', 'B', etc]
```

The idea is you can create a Universe without using files, and it'll make the tests faster if we don't (and also makes the tests not reliant on TPRParser working)

Reply:
thanks. I went with the 2nd choice:
```
def test_segmentgroup_pickle():
    u = mda.Universe.empty(10)
    u.add_Segment(segid="X")
    u.add_Segment(segid="Y")
    u.add_Segment(segid="Z")
    segids = ["A", "X", "Y", "Z"]
    u.add_TopologyAttr("segids", values=["A", "X", "Y", "Z"])
    seg_group = mda.SegmentGroup((1, 3), u)
    seg = pickle.loads(pickle.dumps(seg_group))
    assert_equal(seg.universe.segments.segids, segids)
```